plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveConnectorFactory.java

```diff
@@ -164,4 +156,51 @@ public static Connector createConnector(
                     classLoader);
         }
     }
+
+    private static Bootstrap createBootstrap(
+            String catalogName,
+            Map<String, String> config,
+            ConnectorContext context,
+            Module module,
+            Optional<HiveMetastore> metastore,
+            Optional<TrinoFileSystemFactory> fileSystemFactory,
+            boolean quietBootstrap)
+    {
+        Bootstrap app = new Bootstrap(
+                new CatalogNameModule(catalogName),
+                new MBeanModule(),
+                new ConnectorObjectNameGeneratorModule("io.trino.plugin.hive", "trino.plugin.hive"),
+                new JsonModule(),
+                new TypeDeserializerModule(context.getTypeManager()),
+                new HiveModule(),
+                new HiveMetastoreModule(metastore),
+                new HiveSecurityModule(),
+                fileSystemFactory
+                        .map(factory -> (Module) binder -> binder.bind(TrinoFileSystemFactory.class).toInstance(factory))
+                        .orElseGet(() -> new FileSystemModule(catalogName, context.getNodeManager(), context.getOpenTelemetry(), false, quietBootstrap)),
+                new HiveProcedureModule(),
+                new MBeanServerModule(),
+                binder -> {
+                    binder.bind(OpenTelemetry.class).toInstance(context.getOpenTelemetry());
+                    binder.bind(Tracer.class).toInstance(context.getTracer());
+                    binder.bind(NodeVersion.class).toInstance(new NodeVersion(context.getNodeManager().getCurrentNode().getVersion()));
+                    binder.bind(NodeManager.class).toInstance(context.getNodeManager());
+                    binder.bind(VersionEmbedder.class).toInstance(context.getVersionEmbedder());
+                    binder.bind(MetadataProvider.class).toInstance(context.getMetadataProvider());
+                    binder.bind(PageIndexerFactory.class).toInstance(context.getPageIndexerFactory());
+                    binder.bind(PageSorter.class).toInstance(context.getPageSorter());
+                    binder.bind(CatalogName.class).toInstance(new CatalogName(catalogName));
+                },
+                binder -> newSetBinder(binder, SessionPropertiesProvider.class).addBinding().to(HiveSessionProperties.class).in(Scopes.SINGLETON),
+                module);
+
+        if (quietBootstrap) {
+            app.quiet()
```

Comment:
it would be better to have a single method that accepts boolean flag as these are always called together in the if

Reply:
this will allow for the chained call without the need to assign to a variable

Reply:
let's add `withQuiet(boolean flag)` and `withSkipErrorReporting(boolean flag)`

Reply:
Yes. Inconsistent? Maybe, but more readable in the callsite.

Reply:
For consistency we can add all methods with boolean flags

Reply:
I don't have a strong opinion on this. However, the construct with conditionally applying `quiet` and `skipErrorReporting` is currently used in five places, and I don't think it will be needed for other connectors. In fact, I can replace four of these five usages with a different approach - it will be a chained call.

It seems to me that introducing a new convention only for this case might be overkill, especially considering that `Bootstrap` is used extensively elsewhere.